pyomo/contrib/doe/tests/test_fim_doe.py

```diff
@@ -26,67 +26,354 @@
+        var_T = 'T'
+        indices_T = {0: t_control}
+        exp1_T = [470, 300, 300, 300, 300, 300, 300, 300, 300]
+
+        exp_design.add_variables(
+            var_T,
+            indices=indices_T,
+            time_index_position=0,
+            values=exp1_T,
+            lower_bounds=300,
+            upper_bounds=700,
+        )
+
         self.assertEqual(
-            measure_class.flatten_measure_timeset['C_index_CB'], [0, 0.25, 0.5, 0.75, 1]
+            exp_design.name,
+            [
+                'CA0[0]',
+                'T[0]',
+                'T[0.125]',
+                'T[0.25]',
+                'T[0.375]',
+                'T[0.5]',
+                'T[0.625]',
+                'T[0.75]',
+                'T[0.875]',
+                'T[1]',
+            ],
         )
+        self.assertEqual(exp_design.variable_names_value['CA0[0]'], 5)
+        self.assertEqual(exp_design.variable_names_value['T[0]'], 470)
+        self.assertEqual(exp_design.upper_bounds['CA0[0]'], 5)
+        self.assertEqual(exp_design.upper_bounds['T[0]'], 700)
+        self.assertEqual(exp_design.lower_bounds['CA0[0]'], 1)
+        self.assertEqual(exp_design.lower_bounds['T[0]'], 300)
 
-        # test subset feature
-        subset_1 = {'C': {'CA': t_measure_cb}}
-        measure_subclass1 = Measurements(subset_1)
-        self.assertTrue(measure_class.check_subset(measure_subclass1))
+        exp1 = [4, 600, 300, 300, 300, 300, 300, 300, 300, 300]
+        exp_design.update_values(exp1)
+        self.assertEqual(exp_design.variable_names_value['CA0[0]'], 4)
+        self.assertEqual(exp_design.variable_names_value['T[0]'], 600)
 
 
 class TestParameter(unittest.TestCase):
-    """Test the parameter class"""
+    """Test the ScenarioGenerator class, generate_scenario function."""
 
     def test_setup(self):
         # set up parameter class
         param_dict = {'A1': 84.79, 'A2': 371.72, 'E1': 7.78, 'E2': 15.05}
 
-        scenario_gene = Scenario_generator(param_dict, formula='central', step=0.1)
-        parameter_set = scenario_gene.simultaneous_scenario()
+        scenario_gene = ScenarioGenerator(
+            param_dict, formula=finite_difference_step.central, step=0.1
+        )
+        parameter_set = scenario_gene.scenario_data
+
+        self.assertAlmostEqual(parameter_set['eps-abs']['A1'], 16.9582, places=1)
+        self.assertAlmostEqual(parameter_set['eps-abs']['E1'], 1.5554, places=1)
+        self.assertEqual(parameter_set['scena_num']['A2'], [2, 3])
+        self.assertEqual(parameter_set['scena_num']['E1'], [4, 5])
+        self.assertAlmostEqual(parameter_set['scenario'][0]['A1'], 93.2699, places=1)
+        self.assertAlmostEqual(parameter_set['scenario'][2]['A2'], 408.8895, places=1)
+        self.assertAlmostEqual(parameter_set['scenario'][-1]['E2'], 13.54, places=1)
+        self.assertAlmostEqual(parameter_set['scenario'][-2]['E2'], 16.55, places=1)
+
+
+class TestSpecialSet(unittest.TestCase):
```

Comment:
Check all of the test names. Update to reflect new API. We no longer have a SpecialSet.